### ISSUE TITLE
Cue box's size would be incorrectly changed when setting 'align:start' or 'align:end'

### ISSUE THREAD

**Issue by alastor0325** [w3c/webvtt#458]
In current spec, [`ComputedPosition`](https://www.w3.org/TR/webvtt1/#cue-computed-position) only takes the text alignment `left` and `right` into account, but it should consider the text alignment `start` and `end` as well. Otherwise, it would result in incorrect cue box size.

For example,
If user set `align:start`, the `ComputedPosition` would be `50`, which is its default value.  Assume the text is LTR, so the [`computed position alignment`](https://www.w3.org/TR/webvtt1/#cue-computed-position-alignment) would be `line-left`. That makes the cue box's size become `50` according to the spec [7.2.2](https://www.w3.org/TR/webvtt1/#ref-for-cue-computed-position-2). 

As a result, the cue box's left side would be aligned to the center of the video rendering area. However, I think the correct behavior is to make the cue box's size to `100` and align the cue box's left side to the left side of video rendering area, like what we've done in [`align_start-ref.html`](http://w3c-test.org/webvtt/rendering/cues-with-video/processing-model/align_start-ref.html) or [`align_end-ref.html`](http://w3c-test.org/webvtt/rendering/cues-with-video/processing-model/align_end-ref.html).

I think the simple solution is that to check `computed position alignment` in the algorithm for `computed position`. If `computed position alignment` is `line-left`, then return `0`; if `computed position alignment` is `line-right`, then return `100`.

**Comment by gkatsev** [comment]
This makes total sense to me. Text alignment's `start` and `end` values match `left` and `right` depending on the language direction and the computed position alignment already tells us how to interpret those 4 values.
And I guess this covers the case when position alignment is set explicitly too? For example, programmatically doing `cue.positionAlign = 'line-left'`.

**Comment by alastor0325** [comment]
> And I guess this covers the case when position alignment is set explicitly too? For example, programmatically doing `cue.positionAlign = 'line-left'`.

Agree. So considering `computed position alignment` would be a good choice for us because it would consider either the result of  `position alignment` user set explicitly or the base direction of the cue text.

**Comment by nickasd** [comment]
I'm also quite confused by the fact that in the specification `position alignment` takes text alignment `start` end `end` into account, but `position` treats those cases as if they were `center`. Setting `position` to `auto` and `align` to `start` or `end` causes the cue text to start at the center and expand towards a viewport edge, which looks quite weird. It would also make more sense to me to have it start at the viewport edge instead.

**Comment by nickasd** [comment]
It also seems that there is a contradiction between the specification of `computed position alignment` and what the second green box under the `computed position` specification states:

> Even for horizontal cues with right-to-left cue text, the cue box is positioned from the left edge of the video viewport. This allows defining a rendering space template which can be filled with either left-to-right or right-to-left cue text, or both.

According to that, it would make sense to treat `start` and `end` like absolute values, regardless of the text direction. Currently that absolute value would be `center`, although `left` and `right` would probably make more sense. But then it would also make sense to have a similar behaviour for `position alignment`, which doesn't define a "rendering space template which can be filled with either left-to-right or right-to-left cue text, or both", since it causes the box to change position depending on the text direction.
